### PR TITLE
[video_player_avplay] Support ADAPTIVE_INFO property

### PR DESCRIPTION
**Pull request by xiaowei-guan**
```dart
    _controller = VideoPlayerController.network(
      'https://test.playready.microsoft.com/smoothstreaming/SSWSS720H264PR/SuperSpeedway_720.ism/Manifest',
      streamingProperty: {'ADAPTIVE_INFO':'BITRATES=5000~10000|STARTBITRATE=HIGHEST|SKIPBITRATE=LOWEST'}
    );
```
Add streamingProperty at the time of initialization, only support ADAPTIVE_INFO now.
More detail, please refer to:
[Playing Adaptive Streaming Media](https://developer.samsung.com/smarttv/develop/guides/multimedia/adaptive-streaming.html)

### CODE REVIEW

File: packages/video_player_avplay/lib/video_player.dart [hunk 1]
```diff
@@ -308,6 +312,11 @@ class VideoPlayerController extends ValueNotifier<VideoPlayerValue> {
   /// Only for [VideoPlayerController.network].
   final Map<String, dynamic>? playerOptions;
 
+  /// Sets specific feature values for HTTP, MMS, or specific streaming engine (Smooth Streaming, HLS, DASH, DivX Plus Streaming, or Widevine).
+  /// The available streaming properties depend on the streaming protocol or engine.
+  /// Only for [VideoPlayerController.network].
+  final Map<String, dynamic>? streamingProperty;
```
Comment (JSUYA) [inline packages/video_player_avplay/lib/video_player.dart:318]: Are there any plan for additional property types besides `ADAPTIVE_INFO`?
According to https://developer.samsung.com/smarttv/develop/api-references/samsung-product-api-references/avplay-api.html, String type seems to be sufficient.

What about `Map<String, String>`?

**Review by JSUYA — COMMENTED** [review]
I added ADAPTIVE_INFO to mpd type streaming (DRM Widevine) and tested it.
In fact, I couldn't find any changes. Can you recommend a more appropriate sample?


```
--- a/packages/video_player_avplay/example/lib/main.dart
+++ b/packages/video_player_avplay/example/lib/main.dart
@@ -252,6 +252,10 @@ class _DrmRemoteVideoState extends State<_DrmRemoteVideo> {

     _controller = VideoPlayerController.network(
       'https://storage.googleapis.com/wvmedia/cenc/hevc/tears/tears.mpd',
+      streamingProperty: {
+        'ADAPTIVE_INFO':
+            'BITRATES=5000~10000|STARTBITRATE=HIGHEST|SKIPBITRATE=LOWEST'
+      },
       drmConfigs: DrmConfigs(
         type: DrmType.widevine,
         licenseCallback: (Uint8List challenge) async {

```

File: packages/video_player_avplay/lib/video_player.dart (hunk 1 above)
Comment (xiaowei-guan) [reply packages/video_player_avplay/lib/video_player.dart]: Ok

**Comment by xiaowei-guan** [comment]
> I added ADAPTIVE_INFO to mpd type streaming (DRM Widevine) and tested it. In fact, I couldn't find any changes. Can you recommend a more appropriate sample?
> 
> ```
> --- a/packages/video_player_avplay/example/lib/main.dart
> +++ b/packages/video_player_avplay/example/lib/main.dart
> @@ -252,6 +252,10 @@ class _DrmRemoteVideoState extends State<_DrmRemoteVideo> {
> 
>      _controller = VideoPlayerController.network(
>        'https://storage.googleapis.com/wvmedia/cenc/hevc/tears/tears.mpd',
> +      streamingProperty: {
> +        'ADAPTIVE_INFO':
> +            'BITRATES=5000~10000|STARTBITRATE=HIGHEST|SKIPBITRATE=LOWEST'
> +      },
>        drmConfigs: DrmConfigs(
>          type: DrmType.widevine,
>          licenseCallback: (Uint8List challenge) async {
> ```

You can test it with below codes:
```dart
            _NetworkVideo(
              title: '0~5247438',
              url:
                  'https://cph-p2p-msl.akamaized.net/hls/live/2000341/test/master.m3u8',
              adaptiveInfo: 'BITRATES=0~5247438',
            )
```
You can refer to this sample code:
https://github.com/xiaowei-guan/plugins/tree/avplay_set_streaming_property_sample_code/packages/video_player_avplay

**Review by hyue7 — APPROVED** [review]

**Review by JSUYA — APPROVED** [review]

**Pull request merged**